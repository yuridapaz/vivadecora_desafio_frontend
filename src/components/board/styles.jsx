import styled from 'styled-components';

const handleBoardSize = (boardLevel) => {
  switch (boardLevel) {
    case 'easy':
      return 'grid-template-columns: repeat(2, 1fr);grid-template-rows: repeat(2, 1fr);';
    case 'medium':
      return 'grid-template-columns: repeat(3, 1fr);grid-template-rows: repeat(3, 1fr);';
    case 'hard':
      return 'grid-template-columns: repeat(4, 1fr);grid-template-rows: repeat(4, 1fr);';
  }
};

const handleBoardNaming = (boardLevel) => {
  switch (boardLevel) {
    case 'easy':
      return "content: 'Fácil'";

    case 'medium':
      return "content: 'Médio'";

    case 'hard':
      return "content: 'Difícil'";
  }
};

export const StyledBoardComponent = styled.div`
  color: ${({ theme }) => theme.colors.white};
  width: 300px;
  height: 400px;

  .upper_content {
    display: flex;
    justify-content: space-between;
    padding: 0 20px;
    font-size: ${({ theme }) => theme.fontSize.lg};
    font-weight: 700;
    .restart_icon {
      background: none;
      outline: none;
      border: none;
      font-size: ${({ theme }) => theme.fontSize.xl};
      cursor: pointer;
    }
  }

  .board_container {
    width: 300px;
    height: 300px;
    padding: ${({ theme }) => theme.spacing.xxs};
    background-color: ${({ theme }) => theme.colors.white};
    border-radius: ${({ theme }) => theme.borderRadius.xs};
    margin: 12px auto;
    box-shadow: ${({ theme }) => theme.boxShadow};
    display: grid;
    ${({ boardLevel }) => handleBoardSize(boardLevel)}
    gap: ${({ theme }) => theme.spacing.xxs};
  }

  .difficulty {
    text-align: center;
    margin-top: ${({ theme }) => theme.spacing.md};
    span {
      background-color: ${({ theme }) => theme.colors.white};
      padding: ${({ theme }) => theme.spacing.xxxs};
      border-radius: ${({ theme }) => theme.borderRadius.xs};
      color: ${({ theme }) => theme.colors.red};
      font-weight: 600;

      &::before {
        ${({ boardLevel }) => handleBoardNaming(boardLevel)};
      }
    }
  }
`;
