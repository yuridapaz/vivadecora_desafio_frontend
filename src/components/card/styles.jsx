import styled from 'styled-components';

const CardComponentS = styled.div`
  border-radius: 5px;
  cursor: pointer;
  position: relative;

  .front {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100%;
    height: 100%;
    border-radius: 5px;
    font-size: 3.5rem;
    position: absolute;
    transform: rotateY(90deg);
    transition: all ease-in 0.2s;
  }

  &.flipped .front {
    transform: rotateY(0deg);
    transition-delay: 0.2s;
  }

  .matched {
    background-color: #52bd52;
  }

  .wrong {
    background-color: #ff5656;
  }

  .back {
    width: 100%;
    height: 100%;
    /* border-radius: ${({ theme }) => theme.borderRadius.xs}; */
    background: linear-gradient(135deg, #eceddc 25%, transparent 25%),
      linear-gradient(225deg, #eceddc 25%, transparent 25%),
      linear-gradient(315deg, #eceddc 25%, transparent 25%),
      linear-gradient(45deg, #eceddc 25%, transparent 25%);
    background-size: 30px 30px;

    transform: rotateY(0deg);
    transition: all ease-in 0.2s;
    transition-delay: 0.2s;
  }

  &.flipped .back {
    transform: rotateY(90deg);
    transition-delay: 0s;
  }
`;

export default CardComponentS;
