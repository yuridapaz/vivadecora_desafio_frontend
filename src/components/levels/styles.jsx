import styled from 'styled-components';

const LevelContainer = styled.div`
  width: 80%;
  max-width: 400px;
  height: 400px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  .board_link {
    align-self: center;
    text-decoration: none;
  }
`;

export default LevelContainer;
