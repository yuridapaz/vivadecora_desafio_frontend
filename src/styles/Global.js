import { createGlobalStyle } from 'styled-components';

const GlobalStyle = createGlobalStyle`

* {
   margin: 0;
  padding: 0;
  box-sizing: border-box;
}

body {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-family: ${({ theme }) => theme.fonts.roboto};
}

.app {
  min-height: 100vh;
  height: 100%;
  width: 100vw;
  background: linear-gradient(-45deg, #ff1c1c, #ff5656);
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.header {
  margin-bottom: 30px;
}

`;

export default GlobalStyle;
