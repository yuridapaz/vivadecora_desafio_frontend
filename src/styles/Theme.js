export const theme = {
  colors: {
    primaryTextColor: '#594F43',
    secondaryTextColor: '#777777',
    black: '#000000',
    white: '#fff',
    red: '#ff1c1c',
    background: {
      appBackground: 'linear-gradient(-45deg, #ff1c1c, #ff5656)',
      boardBackground: '#fff',
      cardCurrentBackground: 'blue',
      cardCorrectBackground: '#52bd52',
      cardWrongBackground: '#ff5656',
      // Manter em uma Ãºnica linha //
      cardFlippedBackground:
        'linear-gradient(135deg, #eceddc 25%, transparent 25%),linear-gradient(225deg, #eceddc 25%, transparent 25%),linear-gradient(315deg, #eceddc 25%, transparent 25%),linear-gradient(45deg, #eceddc 25%, transparent 25%)',
    },
  },
  boxShadow: '4px 4px #000000b4',
  borderRadius: {
    xs: '4px',
    sm: '8px',
    md: '12px',
    lg: '16px',
  },
  fonts: {
    roboto: " 'Roboto', sans-serif ",
  },
  fontSize: {
    xxxs: '0.5rem',
    xxs: '0.6rem',
    xs: '0.7rem',
    sm: '0.8rem',
    md: '1rem',
    lg: '1.2rem',
    xl: '1.5rem',
    xxl: '2rem',
    xxxl: '2.5rem',
  },
  spacing: {
    xxxs: '4px',
    xxs: '8px',
    xs: '16px',
    sm: '24px',
    md: '32px',
    lg: '40px',
    xl: '48px',
    xxl: '56px',
    xxxl: '64px',
  },
};
