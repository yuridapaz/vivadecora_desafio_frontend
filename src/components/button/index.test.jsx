import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { describe, test, vi } from 'vitest';

import { ButtonStyled } from '.';
import { theme } from '../../styles/Theme';

describe('Button', () => {
  beforeEach(() => {
    render(
      <ButtonStyled theme={theme} data-testid='button-test-id'>
        Texto
      </ButtonStyled>
    );
  });

  test('Should be able to render based on the children prop', () => {
    expect(screen.getByTestId('button-test-id')).toHaveTextContent('Texto');
  });
});
