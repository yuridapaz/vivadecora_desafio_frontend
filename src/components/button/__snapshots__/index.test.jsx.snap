// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button with Styled Components > Should be able to present styled components properties 1`] = `
.c0 {
  width: 250px;
  padding: 16px;
  margin: 16px;
  border: none;
  border-radius: 12px;
  box-shadow: 4px 4px #000000b4;
  color: #000000;
  font-size: 1.2rem;
  font-weight: 600;
}

.c0:hover {
  cursor: pointer;
}

<button
  className="c0"
  data-testid="button-test-id"
>
  Texto
</button>
`;
